iree/compiler/Conversion/LinalgToLLVM/LinalgBufferizePass.cpp

```diff
@@ -0,0 +1,447 @@
+                                        Operation *op, Value rankedTensor,
+                                        IREE::HAL::InterfaceBindingOp bindingOp,
+                                        bool typeErase = false) {
+  assert(bindingOp);
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPointToStart(
+      &op->getParentOfType<FuncOp>().getBlocks().front());
+
+  // Get the corresponding memref type from the tensor type.
+  auto tensorType = rankedTensor.getType().cast<RankedTensorType>();
+  auto bufferType = getMemrefTypeForTensor(tensorType);
+
+  if (typeErase)
+    // HAL tensor tile ops does not have enough info to reconstruct the original
+    // buffer type. So we just type erase it for now.
+    // TODO: evolve to target flow ops directly with additional shape annotation
+    // so we do not need to type erase.
+    bufferType = MemRefType::get(
+        SmallVector<int64_t>(bufferType.getRank(), ShapedType::kDynamicSize),
+        bufferType.getElementType(), bufferType.getAffineMaps(),
+        bufferType.getMemorySpace());
+  // Create the placeholder op for the backing buffer. Make sure shape
+  // annotation is carried over if exists.
+  auto phOp =
+      b.create<IREE::PlaceholderOp>(loc, bufferType, "interface buffer");
+  phOp.setAttr(getBindingAttrName(), getBinding(op));
+  StringRef attrName = getOperandResultNumAttrName();
+  if (auto operandResultNumAttr = op->getAttr(attrName))
+    phOp.setAttr(attrName, operandResultNumAttr);
+
+  return phOp;
+}
+
+// TODO: evolve into flow.interface.load.tensor when the abstraction to get
+// a raw void* / memref<?xi8> + offset + shape exists.
+// Then we can immediately emit a view op.
+// TODO: canonicalizations of the view op.
+LogicalResult convertInterfaceLoadTensorOp(
+    OpBuilder &b, IREE::HAL::InterfaceLoadTensorOp loadOp,
+    BlockAndValueMapping &bvm) {
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPoint(loadOp);
+
+  // TODO: view + subview
+  if (!matchPattern(loadOp.offset(), m_Zero())) {
+    return loadOp.emitError("unhandled non-zero offset");
+  }
+
+  // Potentially duplicate placeholders will be taken care of by CSE.
+  auto phOp = createPlaceholderOp(b, loadOp->getLoc(), loadOp, loadOp.result(),
+                                  loadOp.queryBindingOp());
+  Value buffer = phOp.getResult();
+  bvm.map(loadOp.result(), buffer);
+  return success();
+}
+
+// TODO: evolve into flow.interface.load.tensor.tile when the abstraction
+// exists. This will turn into some subview chain from the base view.
+LogicalResult convertInterfaceLoadTensorOp(
+    OpBuilder &b, IREE::HAL::InterfaceLoadTensorTileOp loadOp,
+    BlockAndValueMapping &bvm) {
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPoint(loadOp);
+
+  // TODO: view + subview
+  if (!matchPattern(loadOp.base_offset(), m_Zero())) {
+    return loadOp.emitError("unhandled non-zero offset");
+  }
+
+  // Potentially duplicate placeholders will be taken care of by CSE.
```

Comment:
The comment is not really true. I dont think they will. On the SPIR-V side they are handled as effectively "aliased" global variables. Dont know off the top of my head what happens on the CPU side. Just drop the comment for now.

Reply:
Dropping the comment, this will be true once we go from flow to view ops but is premature for now indeed.